src/meteodatalab/operators/vertical_extrapolation.py

```diff
@@ -0,0 +1,114 @@
+"""Vertical extrapolation operators."""
+
+# Third-party
+import numpy as np
+import xarray as xr
+
+# Local
+from .. import physical_constants as pc
+
+LAPSE_RATE = 0.0065  # K m^-1
+H1 = 2000.0
+H2 = 2500.0
+T1 = 298.0
+
+
+def extrapolate_temperature_sfc2p(
+    t_sfc: xr.DataArray,
+    z_sfc: xr.DataArray,
+    p_sfc: xr.DataArray,
+    p_target: float,
+) -> xr.DataArray:
+    """Extrapolate temperature to a target pressure level.
+
+    Implements the algorithm described in [1]_. The algorithm extrapolates
+    temperature from the surface to a target pressure level using a
+    polynomial expression of a dimensionless variable y, which is a function of
+    the surface temperature, surface pressure, and geopotential. It assumes
+    a constant lapse rate of 0.0065 K m^-1.
+
+    Parameters
+    ----------
+    t_sfc : xr.DataArray
+        Surface temperature [K].
+    z_sfc : xr.DataArray
+        Surface geopotential.
+    p_sfc : xr.DataArray
+        Surface pressure.
+    p_target : float
+        Target pressure level.
+
+    Returns
+    -------
+    xr.DataArray
+        Extrapolated temperature at the target pressure level.
+
+    References
+    ----------
+    .. [1] https://www.umr-cnrm.fr/gmapdoc/IMG/pdf/ykfpos46t1r1.pdf
+
+    """
+    y = _vertical_extrapolation_y_term(t_sfc, p_sfc, z_sfc, p_target)
+    return t_sfc * (1 + y + (y**2) / 2 + (y**3) / 6)
```

Comment:
This looks like a Taylor series for the exponential truncated to the 3rd order so I expect this to be accurate within some bounds regarding the value of `p_target`. Is there any guidance in the reference that would help to put in a sanity check?